src/karapace/schema_registry.py

```diff
@@ -465,3 +448,48 @@ def send_delete_subject_message(self, subject: Subject, version: Version) -> Non
         key = {"subject": subject, "magic": 0, "keytype": "DELETE_SUBJECT"}
         value = {"subject": subject, "version": version.value}
         self.producer.send_message(key=key, value=value)
+
+    def check_schema_compatibility(
+        self,
+        new_schema: ValidatedTypedSchema,
+        subject: Subject,
+    ) -> SchemaCompatibilityResult:
+        result = SchemaCompatibilityResult(SchemaCompatibilityType.compatible)
+
+        compatibility_mode = self.get_compatibility_mode(subject=subject)
+        all_schema_versions: dict[Version, SchemaVersion] = self.database.find_subject_schemas(
+            subject=subject, include_deleted=True
+        )
+        live_versions: list[Version] = self.get_live_versions_sorted(all_schema_versions)
```

Comment:
nit: do we need the explicit type declaration here? 🤔
It should inferred by the return type of the function